lib/inspec/metadata.rb

```diff
@@ -177,27 +177,45 @@ def self.finalize_supports(supports, logger)
       end
     end
 
-    def self.finalize(metadata, profile_id, logger = nil)
+    def self.finalize_name(metadata, profile_id, original_target)
+      # profile_id always overwrites whatever already exists as the name
+      unless profile_id.to_s.empty?
+        metadata.params[:name] = profile_id.to_s
+        return
+      end
+
+      # don't overwrite an existing name
+      return unless metadata.params[:name].nil?
+
+      # if there's a title, there is no need to set a name too
+      return unless metadata.params[:title].nil?
```

Comment:
I'm confused by this comment. We don't actually set the name by `title`. This simply does not set a default `name` if there's already a `title` field because it would be counter-intuitive to our users.

Example, if I had a profile inspec.yml with `title: My Profile`, without this line, we would leave the title alone AND create a `name` containing "tests from tests/foo"... and then you'd have CLI formatter output looking like this:

```
Profile: My Profile (tests from tests/foo)
```

... which might be fine but felt like added functionality out of the scope of this bug fix. Therefore, I opted not to add a `name` if there was already a `title`.